### ISSUE TITLE
Unfortunately Ojo has stopped 

### ISSUE THREAD

**Issue by tarmacalastair** [penguin86/ojo#27]
Hello,

I just found your app on Fdroid recently and it is fantastic and works on my current phone and 2 old phones (Android 5 and 9). I have found if I pair it up with Automation (https://f-droid.org/packages/com.jens.automation2/) then I can use an old device and make Ojo start whenever I turn on the screen and stay on for, say 5 minutes, and then turn off the screen to save the battery.

Then I thought, what a perfect way to make use of an old Android tablet, giving a bigger screen to attach to the wall. Unfortunately my tablet is about 10 years old and runs Android 4.2.2. I started with the current Ojo and have since tried all versions back to 0.0.1 but every time it starts fine but closes with "Unfortunately Ojo has stopped" as soon as I tap the settings button to add my camera. I know that the tablet's hardware can play back my rtsp camera stream because it works in the stock "Gallery" app. But this app is just nothing like as usable as yours.

Just wondered if you had any thoughts that might help? I realise I can't expect any proper help with such an old OS but thought I'd ask in case you know of something obvious.

Thanks
Alastair

**Comment by penguin86** [comment]
Hello,
unfortunately, Ojo requires Andoid 5.0, as you can see in its store page: https://f-droid.org/it/packages/it.danieleverducci.ojo/
BTW, I understand the need to recycle old hardware is real, to avoid producing more waste, so I took a chance to make it working on 4.2. I committed the code in https://github.com/penguin86/ojo/tree/android-4.2 and built an apk for you, you can find it here: https://ichibi.eu/index.php/s/MSKQ3FLAtCA37sy
It doesn't crash, but it doesn't also display anything in the emulator. Unfortunately I have no physical device prior to Android 8 to test it on, so let's hope it works on yours. Otherwise, if it displays an empty black screen, the VLC library used to decode the stream may not work on that android version. In that case there's nothing I can do without deeply changing the app, with the risk of breaking it on other devices...
Can you confirm if it works for you?

Thanks!
Daniele

**Comment by tarmacalastair** [comment]
That is brilliant...it works really well using your newly created apk.

Thanks so much for this. Now I can use the old tablet that's been sitting in a drawer for years.

Thanks for putting in the work, very much appreciated.
Alastair

**Comment by penguin86** [comment]
So is it working? No issues at all? This is a little unexpected, as i could obtain only a black screen in the emulator. I should find another 4.x device to try it on, if it works on most devices, I may merge it to be distributed in F-Droid

**Comment by penguin86** [comment]
Will merge this one and make it available in next release.

**Comment by tarmacalastair** [comment]
Sorry for late response. Yes it works on my tablet. But since I responded a few days ago I have now tested further.
With a single camera Ojo works really well. But when I add further cameras then the video loads initially for each camera but then all the cameras freeze and stop updating. Initially I tried with 4 cameras, then I tried with 2. Ojo itself does not freeze but the cameras stop updating, even if I click on one to make it go full screen. The only way that Ojo works for me on this tablet is with just one single camera loaded in the config.

Further update - the above is for my 4 cameras at 1920x1080p (h264). When I use the sub-stream at only 640x360 then I can see all 4 cameras at once, the video updates and I can click to go full screen for each one.
